src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -271,9 +271,6 @@ public ProgrammingExerciseResource(ProgrammingExerciseRepository programmingExer
             return errorResponse;
         }
 
-        // Only save after checking for errors
-        programmingExerciseService.saveParticipations(programmingExercise);
```

Comment:
This seems like a bug to me - we use the tenplate/solution participations from the gui to update the participations in the database.
The problem is that exercises and participations have a circular structure:
An exercise has a templateParticipation, the templateParticipation the exercise attached to it. This is why the templateParticipation doesn't include it's exercise, setting the entry's exercise id in the participation table to null.

I don't see any reason, why we should save the participation data when updating a programming-exercise.

Reply:
Then we need to reconnect them before saving. I think we save the participations, because an attribute of the participation (e.g. the repo url or the build plan id) could have changed, so please activate this call again and make sure to reconnect the participations with the programming exercises before

Reply:
I understand, I changed the endpoint.
I reattach the exercise to its participations in the endpoint (not in the gui when calling the api), because as far as I see it should not be possible to "unlink" participations from their exercise.